apps/studio/src/components/CmsSidebar/CmsSideNav.tsx

```diff
@@ -0,0 +1,192 @@
+import { Suspense } from "react"
+import { useRouter } from "next/router"
+import {
+  Accordion,
+  AccordionButton,
+  AccordionIcon,
+  AccordionItem,
+  AccordionPanel,
+  Box,
+  Flex,
+  Skeleton,
+  Text,
+  useDisclosure,
+  VStack,
+} from "@chakra-ui/react"
+import { Menu } from "@opengovsg/design-system-react"
+import { ResourceType } from "@prisma/client"
+import { BiData, BiFile, BiFolder, BiHomeAlt } from "react-icons/bi"
+
+import { trpc } from "~/utils/trpc"
+
+export const CmsSideNav = () => {
+  const { onToggle } = useDisclosure()
+
+  return (
+    <Flex flexDir="column">
+      <VStack
+        align="flex-start"
+        spacing="1.5rem"
+        pt="2rem"
+        px="1.25rem"
+        pb="0.75rem"
+        boxShadow="sm"
+      >
+        <Menu>
+          {({ isOpen }) => (
+            <>
+              <Menu.Button
+                w="full"
+                onClick={onToggle}
+                isOpen={isOpen}
+                variant="outline"
+              >
+                Create new
+              </Menu.Button>
+              <Menu.List>
+                <Menu.Item>Create new page</Menu.Item>
+                <Menu.Item>Create new folder</Menu.Item>
+                <Menu.Item>Create new collection</Menu.Item>
+              </Menu.List>
+            </>
+          )}
+        </Menu>
+        <Text textStyle="caption-1">Site content</Text>
+      </VStack>
+      <Box mt="4px" px="1.25rem">
+        {/* TODO: update the resource id here */}
+        <Suspense fallback={<Skeleton />}>
+          <SideNavItem
+            resourceId={null}
+            permalink={"/"}
+            resourceType="Folder"
+          />
+        </Suspense>
+      </Box>
+    </Flex>
+  )
+}
+
+// TODO: Add this in after #426 is merged
+const ICON_MAPPINGS = {
+  [ResourceType.Page]: <BiFile />,
+  [ResourceType.Folder]: <BiFolder />,
+  [ResourceType.Collection]: <BiData />,
+  [ResourceType.CollectionPage]: <BiFile />,
+  [ResourceType.RootPage]: <BiHomeAlt />,
+}
+
+interface SideNavItemProps {
+  permalink: string
+  resourceId: string | null
+  resourceType: ResourceType
+  isDisabled?: boolean
+}
+
+const getResourceType = (
+  resourceType: ResourceType,
+): "pages" | "folders" | "collections" => {
+  if (
+    resourceType === ResourceType.Page ||
+    resourceType === ResourceType.CollectionPage
+  ) {
+    return "pages"
+  }
+
+  if (resourceType === ResourceType.Collection) {
+    return "collections"
+  }
+
+  return "folders"
+}
+
+const SideNavItem = ({
+  resourceType,
+  resourceId,
+  permalink,
+  isDisabled,
+}: SideNavItemProps) => {
+  const [children] = trpc.resource.getChildrenOf.useSuspenseQuery({
+    resourceId,
+  })
+
+  const icon = ICON_MAPPINGS[resourceType]
+  const router = useRouter()
+  const { siteId } = router.query
```

Comment:
i think we should pass in `siteId` as a prop, similarly to `resourceId`. this removes the coupling of this component to a required url route.

Reply:
to probe abit on this (i don't mind the change actually) - this component requires the site id at a minimum to fetch the list of children items and it seems logical to tightly couple! i'll make this change, but i actually thought the coupling here was ideal (can't think of a case where we'll pass `siteId` in without the url route)

Reply:
donzxc